extensions/amp-consent/0.1/test/test-tcf-api-command-manager.js

```diff
@@ -230,6 +247,190 @@ describes.realWin(
           );
         });
       });
+
+      describe('addEventListener', () => {
+        it(
+          'responds to changes in policy manager' +
+            ' only after command has been sent',
+          async () => {
+            data = getData('addEventListener', 'callId');
+            tcfApiCommandManager = new TcfApiCommandManager(mockPolicyManager);
+            const getConsentsSpy = env.sandbox.spy(
+              tcfApiCommandManager,
+              'getTcDataPromises_'
+            );
+
+            // Fake a TcData change
+            tcfApiCommandManager.handleTcDataChange_();
+            await macroTask();
+            expect(getConsentsSpy).to.not.be.called;
+
+            // Fake responding to event
+            tcfApiCommandManager.handleTcfCommand(data, mockWin);
+
+            tcfApiCommandManager.handleTcDataChange_();
+            await macroTask();
+            expect(getConsentsSpy).to.be.calledOnce;
+
+            tcfApiCommandManager.handleTcDataChange_();
+            await macroTask();
+            expect(getConsentsSpy).to.be.calledTwice;
+          }
+        );
+
+        it('sends minimal TcData on TcString change', async () => {
+          callId = 'callId';
+          data = getData('addEventListener', callId);
+          mockTcString = 'newTcString';
+          tcfApiCommandManager = new TcfApiCommandManager(mockPolicyManager);
+
+          // Start listening for changes
+          tcfApiCommandManager.handleTcfCommand(data, mockWin);
+
+          tcfApiCommandManager.handleTcDataChange_();
+          await macroTask();
+
+          const postMessageArgs = mockWin.postMessage.args[0];
+          expect(postMessageArgs[0]).to.deep.equals(
+            getTcfApiReturn(
+              callId,
+              tcfApiCommandManager.getMinimalTcDataForTesting(
+                mockMetadata,
+                mockSharedData,
+                mockTcString,
+                0
+              ),
+              true
+            )
+          );
+        });
+
+        it('does not send TcData when TcString is null or the same', async () => {
+          data = getData('addEventListener', 'callId');
+          mockTcString = 'prevTcString';
+          tcfApiCommandManager = new TcfApiCommandManager(mockPolicyManager);
+          tcfApiCommandManager.currentTcString_ = 'prevTcString';
+          tcfApiCommandManager.handleTcfCommand(data, mockWin);
+
+          // Fake a TcData change
+          tcfApiCommandManager.handleTcDataChange_();
+          await macroTask();
+          expect(mockWin.postMessage).to.not.be.called;
+
+          // No TcString case
```

Comment:
Move this to its own test case.

Reply:
Not done?

Reply:
Oops, forgot to delete the old version when I copy+pasted 🤣